addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1487,18 +1487,30 @@ var SnippetsMenu = Widget.extend({
 
         // Auto-selects text elements with a specific class and remove this
         // on text changes
+        const alreadySelectedElements = new Set();
         this.$document.on('click.snippets_menu', '.o_default_snippet_text', function (ev) {
-            $(ev.target).closest('.o_default_snippet_text').removeClass('o_default_snippet_text');
-            $(ev.target).selectContent();
-            $(ev.target).removeClass('o_default_snippet_text');
+            const el = ev.currentTarget;
+            if (alreadySelectedElements.has(el)) {
+                // If the element was already selected in such a way before, we
+                // don't reselect it. This actually allows to have the first
+                // click on an element to select its text, but the second click
+                // to place the cursor inside of that text.
+                return;
+            }
+            alreadySelectedElements.add(el);
+            $(el).selectContent();
         });
         this.$document.on('keyup.snippets_menu', function () {
```

Comment:
_Question:_

I am not sure to understand the keyup need, is that what you explain in the comment below?

Eg, drag & drop Text snippet, which has 2 `<p/>`, click on the first, the `o_default_snippet_text` is removed.
From there, use keyboard arrow to go into second `<p/>` and type something, the text is changed but the class `o_default_snippet_text` is still there: you can click elsewhere and click on that text again and it will be selected fully as if it was never edited.
The keyup is not listened in this case, right? Then when does it kick in (the keyup)?

Just to be sure to understand

Reply:
Not sure to understand your question: the flow you described is actually fixed thanks to my change, no?

> click on the first, the o_default_snippet_text is removed.

Before my commit yes, not anymore

> the text is changed but the class o_default_snippet_text is still there

Not since my commit, exactly because I fixed this keyup function

> you can click elsewhere and click on that text again and it will be selected fully as if it was never edited.

No since you keyup inside and the class was removed at that point (since this commit, fixing the code typo)

--

Note that any keyup (even those which do not edit the actual text) are considered. I won't be changing that in stable, it could actually make some sense actually (especially since before the class was removed on click x)). To review in the task for master I will be creating.

Reply:
That's weird.. I am sure it actually did not work and the class was still there despite the keyup, best guess is that I was on the wrong tab :grimacing: (regular 15.0 runbot and not your branch) since I always have 2 tabs: one for the PR and one for the regular version (Good old times where DB name included the PR number..).

It's why I was asking if you meant that the keyup was not working in that case.

let's ignore this then ^^